Library/SharedFunctions.swift

```diff
@@ -764,3 +748,47 @@ public func attributedCurrency(withProject project: Project, total: Double) -> N
     superscriptAttributes: checkoutCurrencySuperscriptAttributes()
   )
 }
+
+private func estimatedMinMax(
+  from rewards: [Reward],
+  selectedShippingRule: ShippingRule
+) -> (Double, Double) {
+  var min: Double = 0
+  var max: Double = 0
+
+  rewards.forEach { reward in
+    guard reward.shipping.enabled, let shippingRules = reward.shippingRules else { return }
+
+    /// if the current reward's shipping is unrestricted then
+
+    var shippingRule: ShippingRule?
+
+    /// If the reward's shipping prefernce is Anywhere in the world, us it's first and only shipping rule.
+    /// Else use the rule that  matches the selected shipping rule (from the locations dropdown).
+    shippingRule = reward.shipping.preference == .unrestricted
+      ? shippingRules.first
+      : shippingRules
+      .first(where: { $0.location.country == selectedShippingRule.location.country })
+
+    guard let shipping = shippingRule else { return }
+
+    /// Verify there are estimated amounts greater than 0
+    guard let estimatedMin = shipping.estimatedMin?.amount,
+          let estimatedMax = shipping.estimatedMax?.amount,
+          estimatedMin > 0 || estimatedMax > 0 else {
+      return
+    }
+
+    /// Only update min if the current rewards esitmatedMin is smaller.
+    if min == 0 || estimatedMin < min {
+      min = estimatedMin
+    }
+
+    /// Only update max if the current rewards esitmatedMax is bigger.
+    if max == 0 || estimatedMax > max {
+      max = estimatedMax
+    }
```

Comment:
I think we should be adding the maxes together to get the final estimated max, not just taking the biggest one. Ex if I'm pledging to a reward with estimated shipping and I have an add-on with additional shipping, I expect to pay shipping for both. This is also true for min. It's worth confirming that web and android are calculating these the same way, though; we want the total estimated shipping to be consistent across platforms.

Reply:
great call! you're correct. we should be adding these together